src/Components/Pokedex/Pokedex.js

```diff
@@ -1,50 +1,54 @@
-import React, { useContext } from 'react';
+import React, { useContext, useState } from 'react';
 import PokeDetails from '../PokeDetails/PokeDetails';
 import './Pokedex.scss';
 import { AppContext } from '../../AppContext';
+import Select from 'react-select';
 
 const Pokedex = ({ allPokemon, fetchSinglePokemon}) => {
   const { singlePokemon, setSinglePokemon } = useContext(AppContext);
+  const [isShiny, setIsShiny] = useState(false);
 
-  const createSelectMenu = () => {
-    const pokeNames = allPokemon.map((poke, index) => {
-      return  <option value={poke.url} key={index + 1}>{poke.name}</option>;
+  const createPokeList = () => {
+    const pokeList = allPokemon.map(pokemon => {
+      return {
+        label: pokemon.name,
+        url: pokemon.url
+      };
     });
-    return (
-      <select
-        title="Pokemon List"
-        className='pokemon-list'
-        onChange={(event) => { displaySinglePokemon(event); }}
-      >
-        <option
-          value='choose a pokemon'
-          key={0}
-          selected
-        >
-          Pick A Pokemon
-        </option>
-        {pokeNames}
-      </select>
-    );
+    return pokeList;
   };
 
-  const displaySinglePokemon = (event) => {
-    setSinglePokemon(fetchSinglePokemon(event.target.value));
+
+  const displaySinglePokemon = selectedOption => {
+    const singlePokemon = fetchSinglePokemon(selectedOption.url);
+    setSinglePokemon(singlePokemon);
+  };
+
+  const getRandomPokemon = () => {
+    const randomIndex = Math.floor(Math.random() * allPokemon.length);
+    return allPokemon[randomIndex];
   };
 
   return (
     <section className="pokedex-outline">
       <section className="pokedex-screen">
-        {createSelectMenu()}
-        {singlePokemon.name && <PokeDetails details={singlePokemon}/> }
+        <Select
+          onChange={displaySinglePokemon}
+          options={createPokeList()}
+        />
```

Comment:
That's a pretty crafty package! We'll have to talk more about inner-workings.

Reply:
lots of goodies in that package